### ISSUE TITLE
Authorization middleware

### ISSUE THREAD

**Issue by rgbabaev** [tarantool/cartridge#623]
I suggest to implement chain of authorization handlers in Cartridge's webui. Handlers could be added and removed from any frontend module:

```
window.tarantool_enterprise_core.dispatch(
  'cluster:addAuthMiddleware',
  {
    priority: 1,
    handler: async (ctx, next) => {
      ...
      return next(result);
    }
  }
);
```

Handlers will be called asynchronously in order of priority descending. After finishing work, the handler calls` next` with the result of its work as an argument.
If successful, the handler calls `next(null)`. If result is not null, queue stops.

If the handler needs to return an error, it calls `next` with
```
{
  hideModal: boolean,
  effect?: () => void,
  error?: string
}
```

Related to #592

**Comment by rgbabaev** [comment]
Request for comments @KlonD90

**Comment by KlonD90** [comment]
Add unsub method

**Comment by KlonD90** [comment]
And what should be in ctx? We got some data?

**Comment by artur-barsegyan** [comment]
Implemented in #658
